packages/playground/blueprints/src/lib/steps/run-sql-file.ts

```diff
@@ -0,0 +1,69 @@
+import { StepHandler } from '.';
+
+/**
+ * @inheritDoc runSqlFile
+ * @hasRunnableExample
+ * @example
+ *
+ * <code>
+ * {
+ * 	    "step": "runSqlFile",
+ * 		"path": "/tmp/my-backup.sql"
+ * }
+ * </code>
+ */
+export interface RunSqlFileStep {
+	/**
+	 * The step identifier.
+	 */
+	step: 'runSqlFile';
+	/**
+	 * The SQL file to run.
+	 */
+	path: string;
+}
+
+/**
+ * Run an SQL file.
+ *
+ * @param playground The playground client.
+ * @param path The filepath of the SQL file.
+ */
+export const runSqlFile: StepHandler<RunSqlFileStep> = async (
+	playground,
+	{ path },
+	progress?
+) => {
+	progress?.tracker.setCaption(`Executing SQL File`);
+
+	const docroot = await playground.documentRoot;
+
+	const runPhp = await playground.run({
+		code: `<?php
+		require_once ${JSON.stringify(docroot)} . '/wp-load.php';
+
+		$handle = fopen(${JSON.stringify(path)}, 'r');
+		$buffer = '';
+
+		global $wpdb;
+
+		while ($bytes = fgets($handle)) {
+			$buffer .= $bytes;
+
+			if (substr($buffer, -1, 1) !== "\n") {
+				continue;
+			}
+
+			if (substr($buffer, 0, 2) === '--') {
```

Comment:
If a line starts with `--`, then MySQL isn't going to execute anything on that line. The only way to close the comment would be a newline, so I thought it would be worth saving the step. If you think I should remove this, let me know.

Reply:
I don't know how I feel. Something feels a bit odd though that we're stripping comments that start a line and leaving the others in - that this code is neither 0% or 100% but somewhere in the muddy middle. 🤷‍♂️

Reply:
@dmsnell eventually the [SQLite database integration](https://github.com/WordPress/sqlite-database-integration) will support consuming a SQL dump one query at a time. For now, this PR assumes that there is only a single SQL query per line. With that in mind, I'm not sure why this needs to support comments. Any thoughts @seanmorris?

Reply:
@adamziel  @dmsnell  This file has been removed.

https://github.com/WordPress/wordpress-playground/pull/745/files